lib/semian.rb

```diff
@@ -150,10 +152,12 @@ def to_s
   #
   # Returns the registered resource.
   def register(name, **options)
-    circuit_breaker = create_circuit_breaker(name, **options)
-    bulkhead = create_bulkhead(name, **options)
-
-    resources[name] = ProtectedResource.new(bulkhead, circuit_breaker)
+    @lock.synchronize do
+      return resources[name] if resources[name]
```

Comment:
OH, I see what you mean. Good point. Nice catch. I'll fix that in `retrieve_or_register` tomorrow.